src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java

```diff
@@ -42,7 +42,13 @@ public void accept(JobContext ctx) throws Exception {
             for (UserCommons userCommons : allUserCommons) {
                 User user = userRepository.findById(userCommons.getUserId()).orElseThrow(() -> new RuntimeException("Error calling userRepository.findById(" + userCommons.getUserId() + ")"));
                 ctx.log("User: " + user.getFullName() + ", numCows: " + userCommons.getNumOfCows() + ", cowHealth: " + userCommons.getCowHealth());
-
+                
+                int temp = userCommons.getCowDeaths();
+                calculateCowDeaths(userCommons, ctx);
+                if (userCommons.getCowDeaths() > temp) {
+                    totalCows = 0;
+                }
+                
                 var newCowHealth = calculateNewCowHealthUsingStrategy(cowHealthUpdateStrategy, commons, userCommons, totalCows);
```

Comment:
`calculateCowDeaths` runs before the health is updated, so cows with 0 health won't die until the next time health is updated. Is this supposed to happen?

Reply:
Wait isn't totalCows suppose to just be the amount of cows? The purpose of `calculateCowDeaths` is that when health reaches 0.0, all cows are killed, so NumOfCows = 0, and health is reset to default which is 100.0

Reply:
Right, but health isn't updated until `calculateNewCowHealthUsingStrategy` is called. The code right now is checking if health=0 before health is updated.

Reply:
Oh you right, thank you! Let me fix that right now.